server/data/hmppsRestClient.ts

```diff
@@ -137,18 +137,20 @@ export default class RestClient {
     }
   }
 
-  async stream({ path = null, headers = {} }: StreamRequest = {}): Promise<unknown> {
+  async stream({ path = null, headers = {} }: StreamRequest, username?: string): Promise<unknown> {
+    const token = await this.tokenStore.getAuthToken(username)
+
```

Comment:
How does this know whether to use/supply a) The user's own token, and where is this provided?, or b) A system token with a username, or c) A system token without a username (as required by the CommunityApi) ?

Reply:
Should be addressed now!